src/Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core/Services/AuthManager.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IdentityModel;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Extensions.Msal;
+
+namespace Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core.Services
+{
+    /// <summary>
+    /// Handles the credentials retreival for the user.
+    /// </summary>
+    internal class AuthManager : IAuthManager
+    {
+        private readonly IPublicClientApplication publicClientApplication;
+
+        private const string existingClientIdApproved = "872cd9fa-d31f-45e0-9eab-6e460a02d1f1";
+        private readonly string[] ppeScopes = new string[] { "api://5360327a-4e80-4925-8701-51fa2000738e/user_impersonation" };
+        private readonly string[] devScopes = new string[] { "api://a5880bae-b129-42c5-9d6c-d8de8f305adf/user_impersonation" };
+        private readonly string[] prodScopes = new string[] { "api://7ba8d231-9a00-4118-8a4d-9423b0f0a0f5/user_impersonation" };
+        private readonly string[] usedScopes;
+        private const string AadInstanceUrlFormat = "https://login.microsoftonline.com/{0}/v2.0";
+        private const string msAadTenant = "72f988bf-86f1-41af-91ab-2d7cd011db47"; // GUID for the microsoft AAD tenant;
+        private readonly SemaphoreSlim slimSemaphore;
+        private readonly MsalCacheHelper cacheHelper;
+
+        public AuthManager()
+        {
+            string authorityUrl = string.Format(CultureInfo.InvariantCulture, AadInstanceUrlFormat, msAadTenant);
+            this.publicClientApplication = PublicClientApplicationBuilder
+                .Create(existingClientIdApproved)
+                .WithAuthority(AzureCloudInstance.AzurePublic, msAadTenant)
+                .WithDefaultRedirectUri()
+                .Build();
+
+            StorageCreationProperties storageProperties = new StorageCreationPropertiesBuilder($"{nameof(DevCanvasResultSourceService)}_MSAL_cache_{msAadTenant}.txt", MsalCacheHelper.UserRootDirectory)
```

Comment:
@EasyRhinoMSFT Want to draw some attention to this line, it's responsible for caching the creds to 
```appdata/local/DevCanvasResultSourceService...txt```
Want to make sure this is ok and that there isn't somewhere else we should put it (somewhere in ```AppData\Local\Microsoft``` ? should we start a ```AppData\Local\Microsoft\SarifViewer-VS```?)

Reply:
Hmm, I'm not an auth expert, but this seems odd.  Can't we store the token in the Windows cache?

Reply:
The examples that the MSAL team sent me https://github.com/bgavrilMS/MSAL-SingleAccount-SSO/blob/master/SSO/Program.cs shows how they use the StorageCreationPropertiesBuilder pointed to a user directory too.
When you say Windows cache is there a particular place in Windows you would want to cache to like registry keys?

Reply:
I have very little knowledge here - but my understanding was that Windows had a search-able cache so that anything running on it could potentially be re-used.  If these end up going to anywhere in local cache, then that won't be the case - but our cred doesn't even need that capability.  I think this is fine then.